### PR TITLE
GEODE-5638: Organize/Cleanup the geode-native project structure

### PR DESCRIPTION
**Pull request by echobravopapa**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -35,8 +35,8 @@ if(CMAKE_GENERATOR MATCHES Win64*)
   endif()
 endif()
 
+set(CMAKE_DOTNET_TARGET_FRAMEWORK_VERSION "4.5.2")
 set(DOTNET_TARGET_FRAMEWORK_VERSION "4.5.2")
```
Comment (pivotal-jbarrett) [inline CMakeLists.txt]: This is line should be deleted.

File: tests/cli/NewFwkLib/NewFwkLib.csproj.in [hunk 2]
```diff
@@ -84,7 +84,8 @@
       <HintPath>${NUNIT}\bin\framework\nunit.framework.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <ProjectReference Include="$(CMAKE_BINARY_DIR)\clicache\src\Apache.Geode.vcxproj">
+    <ProjectReference Include="$(CMAKE_BINARY_DIR)\clicache\src\Apache.Geode.vcxproj"> <Project>{03A77323-4CE8-3C6C-87CE-530E8B5559DA}</Project>
```
Comment (pivotal-jbarrett) [inline tests/cli/NewFwkLib/NewFwkLib.csproj.in]: The project GUID should not be in here as each time the project is generated the GUID will change.

File: tests/cli/PdxVersion1Lib/PdxVersion1Lib.csproj.in [hunk 3]
```diff
@@ -106,10 +106,10 @@
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="$(CMAKE_BINARY_DIR)\clicache\src\Apache.Geode.vcxproj">
-      <CopyLocalSatelliteAssemblies>true</CopyLocalSatelliteAssemblies>
-      <ReferenceOutputAssembly>true</ReferenceOutputAssembly>
-    </ProjectReference>
+    <ProjectReference Include="${CMAKE_BINARY_DIR}\clicache\src\Apache.Geode.vcxproj">
+      <Project>{03A77323-4CE8-3C6C-87CE-530E8B5559DA}</Project>
```
Comment (pivotal-jbarrett) [inline tests/cli/PdxVersion1Lib/PdxVersion1Lib.csproj.in]: Remove GUID

File: tests/cli/PdxVersion2Lib/PdxVersion2Lib.csproj.in [hunk 4]
```diff
@@ -103,10 +103,10 @@
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="$(CMAKE_BINARY_DIR)\clicache\src\Apache.Geode.vcxproj">
-      <CopyLocalSatelliteAssemblies>true</CopyLocalSatelliteAssemblies>
-      <ReferenceOutputAssembly>true</ReferenceOutputAssembly>
-    </ProjectReference>
+    <ProjectReference Include="${CMAKE_BINARY_DIR}\clicache\src\Apache.Geode.vcxproj">
+      <Project>{03A77323-4CE8-3C6C-87CE-530E8B5559DA}</Project>
```
Comment (pivotal-jbarrett) [inline tests/cli/PdxVersion2Lib/PdxVersion2Lib.csproj.in]: Remove GUID

File: CMakeLists.txt (hunk 1 above)
Comment (echobravopapa) [reply CMakeLists.txt]: This is a partial conversion to using `CMAKE_DOTNET_TARGET_FRAMEWORK_VERSION` , there are some remaining csproj.in files belonging to `cli` test framework that will be addressed in https://issues.apache.org/jira/browse/GEODE-5818  until then we need both

File: tests/cli/NewFwkLib/NewFwkLib.csproj.in (hunk 2 above)
Comment (mmartell) [reply tests/cli/NewFwkLib/NewFwkLib.csproj.in]: done.

File: tests/cli/PdxVersion1Lib/PdxVersion1Lib.csproj.in (hunk 3 above)
Comment (mmartell) [reply tests/cli/PdxVersion1Lib/PdxVersion1Lib.csproj.in]: done.

File: tests/cli/PdxVersion2Lib/PdxVersion2Lib.csproj.in (hunk 4 above)
Comment (mmartell) [reply tests/cli/PdxVersion2Lib/PdxVersion2Lib.csproj.in]: done.

File: CMakeLists.txt (hunk 1 above)
Comment (mmartell) [reply CMakeLists.txt]: @pivotal-jbarrett how's this look now?

File: tests/cli/PdxVersion1Lib/PdxVersion1Lib.csproj.in [hunk 5]
```diff
@@ -109,7 +109,7 @@
     <ProjectReference Include="$(CMAKE_BINARY_DIR)\clicache\src\Apache.Geode.vcxproj">
       <CopyLocalSatelliteAssemblies>true</CopyLocalSatelliteAssemblies>
       <ReferenceOutputAssembly>true</ReferenceOutputAssembly>
-    </ProjectReference>
+    </ProjectReference>
```
Comment (pivotal-jbarrett) [inline tests/cli/PdxVersion1Lib/PdxVersion1Lib.csproj.in]: Do we really need to add this whitespace here and the next file. If not then these two files would have no changes.

**Pull request merged**